src/iroh.rs

```diff
@@ -474,159 +422,281 @@ impl VeilidIrohBlobs {
         std::fs::write(&temp_path, &cbor_data)?;
-        
-        // Convert the path to an absolute path
+    
+        // Convert the path to an absolute path and upload it to the store
         let absolute_temp_path = std::fs::canonicalize(&temp_path)?;
-
-        // Log: Uploading the CBOR data from file to the store
-        println!("Uploading the CBOR data from file to the store...");
-
-        // Upload the CBOR data to the store using upload_from_path
         let collection_hash = self.upload_from_path(absolute_temp_path).await?;
-
-        // Log: Storing the collection hash in the HashMap
-        println!("Storing the collection hash in the HashMap...");
-
-        // Store the collection hash in the HashMap
-        {
-            let mut map = self.collection_hashes.write().await;
-            map.insert(collection_name, collection_hash);
-        }
-
-        // Log: Collection created successfully
+    
+        // Store the new collection hash using the tag system
+        self.store.set_tag(
+            collection_name.to_string().into(),
+            Some(HashAndFormat::new(collection_hash, BlobFormat::Raw)),
+        ).await?;
+    
         println!("Collection created successfully with hash: {:?}", collection_hash);
-
         Ok(collection_hash)
     }
     
-
-    pub async fn set_file(&self, collection_name: String, path: String, hash: Hash) -> Result<Hash> {
-        // Retrieve the collection's current hash
-        let collection_hash = self.collection_hash(&collection_name).await?;
-    
-        // Read the collection data from the store
-        let collection_data = self.read_bytes(collection_hash).await?;
-        let mut collection: Collection = from_slice(&collection_data)?;
     
-        // Add or update the file in the collection
-        collection.files.insert(path, hash);
     
-        // Serialize the updated collection to CBOR
+    pub async fn set_file(&self, collection_name: &String, path: &String, file_hash: &Hash) -> Result<Hash> {
+         // Retrieve the current collection hash from the tag
+        println!("Attempting to retrieve tag for collection: {}", collection_name);
+
+        // Retrieve the collection hash using `collection_hash` (use this to get the collection)
+        let collection_hash = match self.collection_hash(&collection_name).await {
+            Ok(hash) => hash,
+            Err(e) => {
+                println!("Error retrieving tag for collection: {:?}", e);
+                return Err(anyhow!("Tag retrieval failed for collection: {:?}", e));
+            }
+        };
+
+        println!("Successfully retrieved tag for collection: {}", collection_name);
+
+        // Fetch collection data by its hash
+        let collection_entry = self.store.get(&collection_hash).await?;
+        let mut collection: HashMap<String, Hash> = if let Some(entry) = collection_entry {
+            let mut reader = entry.data_reader();
```

Comment:
Would using `read_bytes` be better here? it can handle the data reader stuff via reading the stream